### PR TITLE
Backpack.tf : A new presence which can be used for all TF2 traders

### PR DESCRIPTION
**Pull request by Ali-Joon**
There shouldn't be any problem with the presence.
If there is, contact me in discord: a＠a#8242

### CODE REVIEW

File: websites/B/backpack.tf/presence.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+const browsingStamp = Math.floor(Date.now() / 1000);
+var title: any;
+presence.on("UpdateData", async () => {
+  const path = window.location.pathname;
+  const host = window.location.hostname;
+  const showButtons = await presence.getSetting("buttons");
+  const presenceData: PresenceData = {
+    largeImageKey: "logo"
+  };
+  console.log("Working...");
```
Comment (ririxidev) [inline websites/B/backpack.tf/presence.ts:13]:
```suggestion
```

File: websites/B/backpack.tf/presence.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  if (document.location.hostname == "backpack.tf") {
+    presenceData.startTimestamp = browsingStamp;
+    console.log("Website Detected...");
```
Comment (ririxidev) [inline websites/B/backpack.tf/presence.ts:16]:
```suggestion
```

File: websites/B/backpack.tf/presence.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+});
+const browsingStamp = Math.floor(Date.now() / 1000);
+var title: any;
+presence.on("UpdateData", async () => {
+  const path = window.location.pathname;
+  const host = window.location.hostname;
+  const showButtons = await presence.getSetting("buttons");
+  const presenceData: PresenceData = {
+    largeImageKey: "logo"
+  };
```
Comment (ririxidev) [inline websites/B/backpack.tf/presence.ts:12]:
```suggestion
  const presenceData: PresenceData = {
    largeImageKey: "logo",
    startTimestamp: browsingStamp
  };
```

File: websites/B/backpack.tf/dist/metadata.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  "logo": "https://i.imgur.com/iHCD7nN.png",
+  "thumbnail": "https://i.imgur.com/uQCICM4.png",
+  "version": "1.0.0",
+  "color": "#757575",
+  "tags": [
+    "website"
+  ],
+  "url": "backpack.tf",
+  "description": {
+    "en": "The PriMiD Presence for Backpack.tf!"
```
Comment (ririxidev) [inline websites/B/backpack.tf/dist/metadata.json:18]: Description should be official description of site.

File: websites/B/backpack.tf/dist/metadata.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    "id": "405028899798384640"
+  },
+  "category": "other",
+  "logo": "https://i.imgur.com/iHCD7nN.png",
+  "thumbnail": "https://i.imgur.com/uQCICM4.png",
+  "version": "1.0.0",
+  "color": "#757575",
+  "tags": [
+    "website"
+  ],
```
Comment (ririxidev) [inline websites/B/backpack.tf/dist/metadata.json:15]: weird tags :/

File: websites/B/backpack.tf/dist/metadata.json [hunk 6]
```diff
@@ -0,0 +1,20 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.3",
+  "service": "Backpack.tf",
+  "author": {
+    "name": "a＠a",
+    "id": "405028899798384640"
+  },
+  "category": "other",
+  "logo": "https://i.imgur.com/iHCD7nN.png",
```
Comment (ririxidev) [inline websites/B/backpack.tf/dist/metadata.json:9]: maybe in more quality?

File: websites/B/backpack.tf/dist/metadata.json [hunk 7]
```diff
@@ -0,0 +1,20 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.3",
```
Comment (ririxidev) [inline websites/B/backpack.tf/dist/metadata.json:2]:
```suggestion
  "$schema": "https://schemas.premid.app/metadata/1.4",
```

**Review by rxri — CHANGES_REQUESTED** [review]
and fix eslint issues (deepscan & codacy)

File: websites/B/backpack.tf/dist/metadata.json (hunk 5 above)
Comment (Ali-Joon) [reply websites/B/backpack.tf/dist/metadata.json:15]: I should have changed the metadata file at last, maybe that's why there are some things are not good enough

**Review by Bas950 — CHANGES_REQUESTED** [review]
upload proof of it working

File: websites/B/backpack.tf/presence.ts [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+  clientId: "850295838361649153"
+}),
+browsingStamp = Date.now(),
+presenceData: PresenceData = {
+    largeImageKey: "logo",
+    startTimestamp: browsingStamp
+};
+presence.on("UpdateData", async () => {
+  const showButtons = await presence.getSetting("buttons");
+  if (document.location.hostname == "backpack.tf") {
```
Comment (ririxidev) [inline websites/B/backpack.tf/presence.ts:11]: there is no need for this `if`

**Review by rxri — CHANGES_REQUESTED** [review]
and fix eslint issues.

**Comment by ririxidev** [comment]
No response or any action from pull request author - closing.

**Pull request closed without merging**